assignment4/Course.py

```diff
@@ -0,0 +1,73 @@
+"""Module for reprsenting an academic course."""
+
+from collections import OrderedDict
+
+
+def date_to_string(date, fmt='%m/%d/%Y'):
+    """Convert date object to string.format
+
+    :param date: (datetime.date, datetime.datetime)
+    :param fmt: str: format of return string
+    :return str:
+    """
+    return date.strftime(format=fmt)
+
+
+class Course:
+    """Class defintion for Course."""
+    def __init__(
+        self, name=None, instructor=None, in_progress=False, units=3, grade=None,
+        start_date=None, end_date=None
+    ):
+        """Initialize Course object.
+
+        :param name: str: name of the course
+        :param instructor: str: instructor full name
+        :param in_progress: bool: Is the course in progress?
+        :param units: int
+        :param grade: (int, float)
+        :param start_date: (datetime.date, datetime.datetime)
+        :param end_date: (datetime.date, datetime.datetime)
+        """
+        self.name = name
+        self.instructor = instructor
+        self.in_progress = in_progress
+        self.units = units
+        self.grade = grade
+        if grade is None and in_progress:
+            self.grade = 'N/A'
+        self.start_date = start_date
+        self.end_date = end_date
+
+        self.grade_points_map = OrderedDict({
+            'A+': 4.0, 'A': 4.0, 'A-': 3.7, 'B+': 3.3, 'B': 3.0, 'B-': 2.7,
+            'C+': 2.3, 'C': 2.0, 'C-': 1.7, 'D': 1.0, 'F': 0.0
+        })
```

Comment:
Since Python 3.6 (guaranteed since Python 3.7), dictionaries are ordered by insertion order, so you don't need an OrderedDict. Not a big deal for small things, but OrderedDicts are a bit less performant.